packages/kit/src/runtime/server/respond.js

```diff
@@ -296,7 +300,15 @@ export async function respond(request, options, manifest, state) {
 			if (is_data_request) {
 				return redirect_json_response(e);
 			} else {
-				return redirect_response(e.status, e.location);
```

Comment:
you can keep the `redirect_response` function and add the headers to it.
```js
const response = redirect_response(..);
add_cookies_to_headers(..);
return response;
```

Reply:
This would only add the cookies to the response.
I wanted to do it this way originally but wasn't sure if only cookies should be added or to include the headers from `setHeaders()` too. 

I also didn't want a `location` header from `setHeaders` to overwrite the thrown redirect location.

Maybe I can extend the `redirect_response()` helper to optionally accept a `headers` option?
I noticed there were similar redirect responses in the file not using the helper just so that it could add other headers to it.